pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/loader/SegmentDirectoryLoader.java

```diff
@@ -42,4 +41,13 @@ SegmentDirectory load(URI indexDir, SegmentDirectoryLoaderContext segmentDirecto
   default void delete(SegmentDirectoryLoaderContext segmentDirectoryLoaderContext)
       throws Exception {
   }
+
+  /**
+   * Based on the zkMetadata's and current segment tier, checks whether or not tier migration is needed
+   * @param zkTier segment's ZKMetadata's tier
+   * @param currentTier Current segment tier
+   */
+  default boolean needsTierMigration(String zkTier, String currentTier) {
```

Comment:
nit: rename zkTier to targetTier to be more generic

Reply:
Ack